pkg/cli/whoami/whoami.go

```diff
@@ -171,19 +205,18 @@ func (o *WhoAmIOptions) Run() error {
 		}
 		fmt.Fprintf(o.Out, "%s\n", consoleUrl)
 		return nil
+	case o.resourcePrinterFunc != nil:
+		// If output format is specified, get the user info and print it in the requested format
+		me, err := o.WhoAmI()
```

Comment:
`WhoAmI` function still prints the username into the terminal. So the output of this function is not an valid json/yaml. I think we can pass a boolean to print username or not.